webhook/validation/validation.go

```diff
@@ -0,0 +1,103 @@
+package validation
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/logger"
+	"github.com/Dynatrace/dynatrace-operator/scheme"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	errorConflictingInfraMonitoringAndClassicNodeSelectors = `
+The DynaKubes specifications for infraMonitoring and classicFullStack are conflicting. 
+If both are enabled, the nodeSelectors of those specifications must not select the same nodes.
+This is due to the infraMonitoring and classicFullStack functionalities being incompatible.
+In general, it is advised to use infraMonitoring together with codeModules instead of classicFullStack.
+`
+)
+
+func AddDynakubeValidationWebhookToManager(manager ctrl.Manager) error {
+	manager.GetWebhookServer().Register("/validate", &webhook.Admission{
+		Handler: newDynakubeValidator(),
+	})
+	manager.GetWebhookServer().Register("/healthz", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	return nil
+}
+
+type dynakubeValidator struct {
+	logger logr.Logger
+	clt    client.Client
+}
+
+// InjectClient implements the inject.Client interface which allows the manager to inject a kubernetes client into this handler
+func (validator *dynakubeValidator) InjectClient(clt client.Client) error {
+	validator.clt = clt
+	return nil
+}
+
+func (validator *dynakubeValidator) Handle(_ context.Context, request admission.Request) admission.Response {
```

Comment:
We have the following in `oneagent_utils.go`:
```go
func validate(cr *dynatracev1alpha1.DynaKube) error {
	var msg []string
	if cr.Spec.APIURL == "" {
		msg = append(msg, ".spec.apiUrl is missing")
	}
	if len(msg) > 0 {
		return errors.New(strings.Join(msg, ", "))
	}
	return nil
}
```
checking this here would nice. (so we could remove it from the oneagent_controller)

Reply:
Do you mean moving the apiUrl validation from oneagent_utils to the new webhook logic?

Reply:
moved functionality